zcash_client_sqlite/src/wallet.rs

```diff
@@ -2768,7 +2957,17 @@ pub(crate) fn store_decrypted_tx<P: consensus::Parameters>(
                 )?;
             }
             TransferType::Incoming => {
-                sapling::put_received_note(conn, output, tx_ref, None)?;
+                let _account_id = sapling::put_received_note(
+                    conn,
+                    params,
+                    output,
+                    tx_ref,
+                    d_tx.mined_height(),
+                    None,
+                )?;
+
+                #[cfg(feature = "transparent-inputs")]
+                receiving_accounts.insert(_account_id, KeyScope::EXTERNAL);
```

Comment:
Given that the caller provides `d_tx`, can we trust that `TransferType::Incoming` is only used when the receiving address is derieved via `KeyScope::EXTERNAL`, or is there a consistency check we need to perform here?

Reply:
It would be a bug if `TransferType` is set incorrectly, which is what this consistency check would be guarding against. I don't think that it's necessary to perform that check everywhere.